src/configuration.ts

```diff
@@ -41,6 +41,36 @@ function fixConfig(ts: TSCommon, config: _ts.ParsedCommandLine) {
   return config;
 }
 
+function readTsConfigExtendForTsNodeRecursively(
+  cwd: string,
+  ts: TSCommon,
+  rawConfig: { [key: string]: any } = {}
+): TsConfigOptions {
+  const configKey = 'ts-node';
+  const recurse = (
+    extending: { [key: string]: any } = {},
+    original: { [key: string]: any } = {}
+  ) => {
+    return readTsConfigExtendForTsNodeRecursively(cwd, ts, {
+      [configKey]: { ...extending[configKey], ...original[configKey] },
+    });
+  };
+
+  const { extends: extendFile, ...config } = rawConfig;
+
+  const result =
+    extendFile === undefined
+      ? filterRecognizedTsConfigTsNodeOptions(config[configKey])
+      : [extendFile]
```

Comment:
Nit: get rid of the `filter().map()` chaining.  We prefer code like `const`, `if()`, `for(const a of b)`, etc.

Reply:
@cspotcode It's not for everyone 🐼. Once you have a look over the logic itself, I'll swap it out.